### PR TITLE
stop asking about surface on private (foot-)paths

### PR DESCRIPTION
**Pull request by legofahrrad**

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddFootwayPartSurface.kt [hunk 1]
```diff
@@ -25,6 +25,7 @@ class AddFootwayPartSurface : OsmFilterQuestType<SurfaceAnswer>() {
             and !note:footway:surface
           )
         )
+        and access !~ private|no
```
Comment (matkoniecz) [inline app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddFootwayPartSurface.kt:28]: what about `access=no foot=yes`  taggings? Or `foot=designated`? (some RoW in England have such tagging from wha)

Comment (legofahrrad) [reply app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddFootwayPartSurface.kt:28]: `and access !~ private|no and (!foot or foot !~ private|no)` would be a better filter then, I guess?

Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddFootwayPartSurface.kt:28]: Watch your brackets, mister! ;-)

Comment (FloEdelmann) [reply app/src/main/java/de/westnordost/streetcomplete/quests/surface/AddFootwayPartSurface.kt:28]: I think it should be

```suggestion
        and (access !~ private|no or foot !~ private|no)
```

**Comment by westnordost** [comment]
Code looks good (also for the other). Did you test it (you can create a debug APK from the actions tab)?

**Comment by legofahrrad** [comment]
Sorry, didn't tested it yet (same for #4434)

**Comment by legofahrrad** [comment]
I built a debug apk but it didn't worked as expected. Maybe I did something wrong?
I tried this at this way: https://www.openstreetmap.org/way/300720572
The quests shouldn't show up, because of `access=no` but there were the quests about the surfaces.

**Comment by mnalis** [comment]
@legofahrrad did you run GitHub Action on correct branch?  When you click `Run workflow` it allows you to choose on which branch do you want to run it. It seems you've maybe run it against the default `master` branch, instead against your `patch-2` branch?

**Comment by legofahrrad** [comment]
tried again with `patch-2`, worked now for footpaths but cyclepath surface quest still shows up. The problem: My cyclepath surface PR is on `patch-1`
Tried once again for `patch-1` and worked as expected :)

**Comment by westnordost** [comment]
So, in short you tested it and it works?

**Comment by legofahrrad** [comment]
Yup, works as expected 👍🏻

**Pull request merged**